cron/daily/96-apply-data-retention-policy.ts

```diff
@@ -18,8 +18,36 @@ enum RetentionPeriod {
   REDUCED = '1 month',
 }
 
-const MODEL_RETENTION_PERIODS = new Map<ModelStatic<any>, RetentionPeriod>([
-  [models.Comment, RetentionPeriod.SENSITIVE],
+type AdditionalConditions = Record<string, any> & { deletedAt?: never };
+
+type ModelRetentionPeriodSettings = [ModelStatic<any>, RetentionPeriod, AdditionalConditions?];
+
+/**
+ * Recursively replaces Sequelize operators with their string representation
+ */
+const stringifySequelizeOperators = (value: AdditionalConditions): any => {
+  const result: Record<string, any> = {};
+
+  if (typeof value === 'object') {
+    for (const symbol of Object.getOwnPropertySymbols(value)) {
+      result[symbol.toString()] = value[symbol as unknown as string];
+    }
+
+    for (const [key, val] of Object.entries(value)) {
+      if (typeof val === 'object') {
+        result[key] = stringifySequelizeOperators(val);
+      } else {
+        result[key] = val.toString();
+      }
+    }
+  }
+
+  return JSON.stringify(result);
+};
+
+const MODEL_RETENTION_PERIODS: ModelRetentionPeriodSettings[] = [
+  [models.Comment, RetentionPeriod.FINANCIAL, { ExpenseId: { [Op.not]: null } }],
+  [models.Comment, RetentionPeriod.SENSITIVE, { ExpenseId: { [Op.is]: null } }],
```

Comment:
I think we should be more explicit about which ids we're expecting to be set instead of querying for "everything that is not an expense comment".

Reply:
I agree, pushing an update in https://github.com/opencollective/opencollective-api/pull/10748